src/main/java/Duke.java

```diff
@@ -1,10 +1,303 @@
+        boolean isProperTask = isTodoTask || isDeadlineTask || isEventTask;
+
+        if (line.equals("list")) {
+            System.out.println("____________________________________________________________");
+            System.out.println("Here are your remaining tasks:");
+
+            for (int i = 1; i <= taskList.size(); i++) {
+                System.out.println(
+                        Integer.toString(i)
+                                + ". "
+                                + taskList.get(i - 1)
+                );
+            }
+            System.out.println("____________________________________________________________\n");
+
+        } else if (line.startsWith("done")) {
+            try {
+                int i = Integer.parseInt(line.substring(5)) - 1;
+                taskList.get(i).markAsDone();
+
+                System.out.println("____________________________________________________________\n"
+                        + "Well done! I've marked this task as done: \n"
+                        + taskList.get(i)
+                        + System.lineSeparator()
+                        + "____________________________________________________________\n"
+                );
+            } catch (NumberFormatException e) {
+                System.out.println("Invalid task number. Please mark a valid task number as done!\n");
+
+            } catch (StringIndexOutOfBoundsException e) {
+                System.out.println("Please mark a task number as done!\n");
+            }
+
+        } else if (isProperTask) {
+            //checks for task commands
+            if (isTodoTask) {
+                String actualTask = line.substring(4).trim();
+                if (actualTask.isBlank()) {
+                    throw new EmptyTaskException();
+                }
+                taskList.add(new Todo(actualTask));
+            } else if (isDeadlineTask) {
+                int indexOfSlash = line.indexOf("/");
+                String actualTask = line.substring(8, indexOfSlash - 1).trim();
+                String deadlineBy = line.substring(indexOfSlash + 4);
+                if (actualTask.isBlank()) {
+                    throw new EmptyTaskException();
+                }
+                if (!line.contains("/by")) {
+                    throw new InvalidCommandException();
+                }
+                taskList.add(new Deadline(actualTask, deadlineBy));
+
+            } else {
+                int indexOfSlash = line.indexOf("/");
+                String actualTask = line.substring(5, indexOfSlash - 1).trim();
+                String eventAt = line.substring(indexOfSlash + 4);
+                if (actualTask.isBlank()) {
+                    throw new EmptyTaskException();
+                }
+                if (!line.contains("/at")) {
+                    throw new InvalidCommandException();
+                }
+                taskList.add(new Event(actualTask, eventAt));
+            }
```

Comment:
same goes here... perhaps improve abstraction and readability by consider one method to each else if?